docs/how-tos/use-multi-format-reader.md

```diff
@@ -0,0 +1,399 @@
+
+I recommend to make the statement even stronger, the choices made in the config file define how semantics from the src and trg (NeXus) side are mapped. 
+
+
+```json
+{
+  "/ENTRY/title": "@eln", 
+  "/ENTRY/USER[user]": {
+    "name":"@eln",
+    "address":"@eln"
+  }, 
+  "/ENTRY/INSTRUMENT[instrument]": {
+    "@version":"@attrs:metadata/instrument/version",
+    "DETECTOR[detector]":{
+      "count_time":"@attrs:metadata/instrument/detector/count_time",
+      "count_time/@units":"@attrs:metadata/instrument/detector/count_time_units"
+    }
+  },
+  "/ENTRY/SAMPLE[sample]": {
+    "name":"@eln",
+    "physical_form":"@eln",
+    "temperature":"@eln",
+    "temperature/@units":"@eln"
+  },
+  "/ENTRY/data": {
+    "@axes":["x_values"],
+    "@signal": "data",
+    "data": "@data:y_values",
+    "x_values": "@data:x_values"
+  }
+}
+```
+We also implement a method for setting the config file in the reader:
+```python
+def set_config_file(self, file_path: str) -> Dict[str, Any]:
+    if self.config_file is not None:
+        logger.info(
+            f"Config file already set. Replaced by the new file {file_path}."
+        )
+    self.config_file = file_path
+  
+    return {}
+```        
+
+## Filling the template from the read-in data
+Finally, after reading in all of the data and metadata as well as designing the config file, we can start filling the template. For this, we must implement functions that are called using the reader's **callbacks**.
+
+We will start with the `@attrs` prefix, associated with the `attrs_callback`. We must implement the `get_attr` mehthod:
+```python
+def get_attr(self, key: str, path: str) -> Any:
+    """
+    Get the metadata that was stored in the main file.
+    """
+    if self.hdf5_data is None:
+        return None
+    
+    return self.hdf5_data.get(path)
+```
+Here, we are making use of the `path`, which is the part of the config value that comes after the `@attrs:` prefix. For example, for the config value `"@attrs:metadata/instrument/version"`, the extracted path is `metadata/instrument/version`, which is also one of the keys of the `self.hdf5_data` dictionary.
+
+For the ELN data, we must implement the `get_eln_data` function that gets called from the `eln_callback` when using the `@eln` prefix:
+```python
+def get_eln_data(self, key: str, path: str) -> Any:
+    """Returns data from the given eln path."""
+    if self.eln_data is None:
+        return None
+    
+    return self.eln_data.get(key)
+```
+Here, we are making use of the fact that we have used `CONVERT_DICT` in the `parse_yml` function above. Thus, the keys of the `self.eln_data` dictionary are exactly the same as those in the config file. Therefore, we can just get this data using the `key` coming from the config file.

```

Comment:
Better explain the content of keys and path above (when introducing the special conept keys). Also check for non-empty path, and use rather than key, to allow for the example using eln paths.

Reply:
First part done. Second part not implemented as discussed.